torch/cuda/amp/grad_scaler.py

```diff
@@ -140,8 +135,8 @@ def _check_scale_growth_tracker(self, funcname) -> Tuple[torch.Tensor, torch.Ten
 
     def _lazy_init_scale_growth_tracker(self, dev):
         assert self._growth_tracker is None, "_growth_tracker initialized before _scale"
-        self._scale = torch.full((1,), self._init_scale, dtype=torch.float32, device=dev)
-        self._growth_tracker = torch.full((1,), self._init_growth_tracker, dtype=torch.int32, device=dev)
+        self._scale = torch.full((), self._init_scale, dtype=torch.float32, device=dev)
```

Comment:
What's your motivation for creating these as 0-element tensors instead of 1-element tensors?  What difference does that make for xla?

Reply:
Because the corresponding `AmpForachNonFiniteCheckAndUnscale` and `AmpUpdateScale` XlaOp could only handle 1-element tensors.

Reply:
> could only handle 1-element tensors.

do you mean "could only handle 0-element tensors?"  otherwise I'm confused, they are 1-element tensors in existing code.

Reply:
Maybe there's some misunderstanding on `1-element tensors` and `0-element-tensors`.

Both `torch.full((1,), ...)` and `torch.full((), ...)` are 1-element tensors, with different shape, one is scalar, the other has shape [1]. Please take a look at the details below. And my XlaOp could only handle the scaler one.

```
>>> a = torch.full((1,), 0)
>>> print(a, a.shape)
tensor([0]) torch.Size([1])
>>> b = torch.full((), 0)
>>> print(b, b.shape)
tensor(0) torch.Size([])
```

Reply:
If GradScaler tests in test_cuda.py are passing, this seems ok.

Reply:
@yaochengji Can you point me to why it only support the scalar one? on pytorch/xla we should support both semantics above.